migration/sql-files/084-codebases-spaceid-url-index.sql

```diff
@@ -0,0 +1,15 @@
+-- Delete duplicate entries of codebases in same space
+-- See here: https://wiki.postgresql.org/wiki/Deleting_duplicates
+DELETE FROM codebases
+WHERE id IN (
+    SELECT id
+    FROM (
+        SELECT id, ROW_NUMBER() OVER (partition BY url, space_id ORDER BY id) AS rnum
+        FROM codebases
```

Comment:
There is a certain chance this will delete a 'deleted_at = null' entry and leave the 'deleted_at = time', which would remove the codebase from the users view.

Please add test case

Reply:
done at https://github.com/fabric8-services/fabric8-wit/pull/1925/commits/5032d25fc0157e6808249ffb342dd504107ba800#diff-2a094cd353ec070302cbd8d38edd110aR4 making sure that the entry that exists is the one we want and others are deleted.